xgcm/test/datasets.py

```diff
@@ -27,6 +27,233 @@
+                np.random.rand(6),
+                {
+                    "standard_name": "vertical coordinate",
+                    "units": "[%]",
+                },
+            ),
+        },
+    ),
+    "sgrid3D": xr.Dataset(
+        {
+            "grid": (
+                (),
+                np.array(1, dtype="int32"),
+                {
+                    "cf_role": "grid_topology",
+                    "topology_dimension": 3,
+                    "node_dimensions": "inode jnode knode",
+                    "volume_dimensions": "icell: inode (padding: low) jcell: jnode (padding: low) kcell: knode (padding: high)",
+                    "node_coordinates": "node_lon node_lat node_alt",
+                },
+            ),
+            "c": (
+                ["time", "kcell", "jcell", "icell"],
+                np.random.rand(1, 10, 10, 20),
+                {
+                    "description": "some concentration",
+                    "grid": "grid",
+                    "location": "volume",
+                },
+            ),
+            "u": (
+                ["time", "knode", "jcell", "inode"],
+                np.random.rand(1, 10, 10, 20),
+                {
+                    "description": "x-velocity",
+                    "units": "m s-1",
+                    "grid": "grid",
+                    "location": "face1",
+                },
+            ),
+            "v": (
+                ["time", "knode", "jnode", "icell"],
+                np.random.rand(1, 10, 10, 20),
+                {
+                    "description": "y-velocity",
+                    "units": "m s-1",
+                    "grid": "grid",
+                    "location": "face2",
+                },
+            ),
+            "w": (
+                ["time", "kcell", "jcell", "icell"],
+                np.random.rand(1, 10, 10, 20),
+                {
+                    "description": "y-velocity",
+                    "units": "m s-1",
+                    "grid": "grid",
+                    "location": "face3",
+                },
+            ),
+        },
+        attrs={"conventions": "sgrid-x.x"},
+        coords={
+            "node_lat": (
+                ["knode", "jnode", "inode"],
+                np.random.rand(10, 10, 20),
+                {
+                    "standard_name": "latitude",
+                    "units": "degree_north",
+                },
+            ),
+            "node_lon": (
+                ["knode", "jnode", "inode"],
+                np.random.rand(10, 10, 20),
+                {
+                    "standard_name": "longitude",
+                    "units": "degree_east",
+                },
+            ),
+            "node_vert": (
+                ["knode", "jnode", "inode"],
+                np.random.rand(10, 10, 20),
+                {
+                    "standard_name": "pressure",
+                    "units": "bar",
+                },
+            ),
+        },
+    ),
```

Comment:
These look like very nice and carefully constructed examples. But are they actually being run? Just adding datasets to this list does not make them run. What does your local workflow look like? How are you verifying that the tests run?

These fixtures are imported as follows

https://github.com/xgcm/xgcm/blob/0934c3722d0b8e4842fe8e1bc3daba2ee70f665b/xgcm/test/test_grid.py#L7-L14

AFAICT, unless the datasets are added to one of the fixtures such as `all_datasets` or `all_2d`, they will not actually run as part of the `test_grid.py` tests.

Reply:
Added the examples (with some changes) into the various sets for running. I know they are now being run because some fail (see discussion below). 🙃